packages/ts/file-router/src/vite-plugin/applyLayouts.ts

```diff
@@ -26,6 +26,36 @@ function enableFlowLayout(route: RouteMeta): RouteMeta {
     : routeWithFlowLayout;
 }
 
+/**
+ * Check if there is a layout available that can handle the given path.
+ * Layouts match the starting parts of the path so '/' will match all paths
+ * and '/home' matches '/home' anything with the start path '/home/*'
+ *
+ * @param layoutPaths - available layout paths
+ * @param path - to check for layout
+ */
+function layoutExists(layoutPaths: string[], path: string) {
+  const splitPath: string[] = path.split('/');
+
+  return (
+    layoutPaths.filter((layout) => {
+      if (layout.length === 0) {
+        return true;
+      }
+      const splitLayout: string[] = layout.split('/');
+      if (splitLayout.length > splitPath.length) {
+        return false;
+      }
+      for (let i = 0; i < splitLayout.length; i++) {
+        if (!(splitPath[i] === splitLayout[i])) {
+          return false;
+        }
+      }
+      return true;
```

Comment:
Isn't this the same as `path.startWith(layout)`?
Another thing, can layouts have patterns e.g. `/home/*`? If so, I believe `/home/*` should match a path like `/home/contacts` but the current logic will return false, if I understand correctly. The trailing `/` and `possible `/*` should also be stripped before this logic is checked.

Reply:
The path parts are split with / so there should be not / in any part. Except if javascript split works in mysterious ways.

No patterns as you only give the path it matches to so all given layout paths are seen as ending with /*

Reply:
Also it is not the same as startWith as if the layout path is `/home/owner` it should not match `/home/owner2/house`, but should match `/home/owner/house`